src/gui/transferlistmodel.cpp

```diff
@@ -295,29 +294,51 @@ BitTorrent::TorrentHandle *TransferListModel::torrentHandle(const QModelIndex &i
 {
     if (!index.isValid()) return nullptr;
 
-    return m_torrents.value(index.row());
+    return m_torrentList.value(index.row());
 }
 
 void TransferListModel::handleTorrentAboutToBeRemoved(BitTorrent::TorrentHandle *const torrent)
 {
-    const int row = m_torrents.indexOf(torrent);
-    if (row >= 0) {
-        beginRemoveRows(QModelIndex(), row, row);
-        m_torrents.removeAt(row);
-        endRemoveRows();
+    const int row = m_torrentMap.value(torrent, -1);
+
+    if (row < 0)
+        return;
+
+    beginRemoveRows({}, row, row);
+
+    m_torrentList.removeAt(row);
+
+    m_torrentMap.remove(torrent);
+    for (int &value : m_torrentMap) {
+        if (value > row)
+            --value;
     }
+
+    endRemoveRows();
 }
 
 void TransferListModel::handleTorrentStatusUpdated(BitTorrent::TorrentHandle *const torrent)
 {
-    const int row = m_torrents.indexOf(torrent);
-    if (row >= 0)
-        emit dataChanged(index(row, 0), index(row, columnCount() - 1));
+    const int row = m_torrentMap.value(torrent, -1);
+
```

Comment:
Did they pay you for the number of lines? IMO, you overuse blank lines in this PR.

Reply:
I just use it whenever I liked it, just point it out where should I remove.

Reply:
This code seems too thin to me.
Besides, in my opinion, you split a logically related things. I would keep together the variable  initialization and further checking it for a valid value.

Reply:
updated.